src/runtime/policy/tests/policy-test.ts

```diff
@@ -435,28 +435,74 @@ policy MyPolicy {
     const policy = manifest.policies[0];
     const schema = policy.targets[0].getMaxReadSchema();
     const expectedSchemas = (await Manifest.parse(`
-      schema Address
-        number: Number
-        street: Text
-        city: Text
-        country: Text
-
-      schema Name
-        last: Text
-
       schema Person
         name: inline Name {last: Text}
-        addresses: List<inline Address {number, street, city}>
+        addresses: List<inline Address {number: Number, street: Text, city: Text}>
       `)).schemas;
     assert.deepEqual(schema, expectedSchemas['Person']);
   });
 
+  it(
+    'Deeply nested schemas are also restricted according to policy', async () => {
+      const manifest = await Manifest.parse(`
+        schema Name
+          first: Text
+          last: Text
+
+        schema Address
+          number: Number
+          street: Text
+          city: Text
+          state: Text
+
+        schema Person
+          name: inline Name
+          address: &Address
+
+        schema Group
+          persons: List<inline Person>
+
+        policy MyPolicy {
+          from Group access {
+            persons {
+              name {
+                last
+              },
+              address {
+                city
+              }
+            }
+          }
+        }`);
+      const maxReadSchemas = IngressValidation.getMaxReadSchemas(manifest.policies);
+      const expectedSchemas = (await Manifest.parse(`
+        schema Name
+          last: Text
+
+        schema Address
+          city: Text
+
+        schema Person
+          name: inline Name
+          address: &Address
+
+        schema Group
+          persons: List<inline Person>
+      `)).schemas;
+      assert.deepEqual(maxReadSchemas['Group'], expectedSchemas['Group']);
+      assert.deepEqual(maxReadSchemas['Address'], expectedSchemas['Address']);
+      assert.isFalse('Person' in maxReadSchemas);
```

Comment:
Why `Person` is not in maxReadSchemas while `Address` is? I don't understand.

Reply:
`Person` and `Name` only appear as inline entities all the schemas. Therefore, they should be directly accessible. That is why they don't appear in `maxReadSchemas`. May be the test should be named as `Only deeply nested references are restricted according to policy`?